code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,354 @@
+		onEliteWon()
+	if(activity == TUMOR_ACTIVE && mychild != null && mychild.stat == DEAD || activity == TUMOR_ACTIVE && QDELETED(mychild))
+		onEliteLoss()
+	
+obj/structure/elite_tumor/proc/onEliteLoss()
+	playsound(loc,'sound/effects/tendril_destroyed.ogg', 200, 0, 50, TRUE, TRUE)
+	visible_message("<span class='boldwarning'>The tumor begins to convulse violently before beginning to dissipate.</span>")
+	if(activity == TUMOR_ACTIVE)
+		visible_message("<span class='boldwarning'>As the tumor closes, something is forced up from down below.</span>")
+		var/obj/structure/closet/crate/necropolis/tendril/lootbox = new /obj/structure/closet/crate/necropolis/tendril(loc)
+		if(boosted)
+			var/lootpick = rand(1, 2)
+			if(lootpick == 1 && mychild.loot_drop != null)
+				new mychild.loot_drop(lootbox)
+			else
+				new /obj/item/tumor_shard(lootbox)
+	qdel(src)
+	
+obj/structure/elite_tumor/proc/onEliteWon()
+	times_won++
+	activity = TUMOR_PASSIVE
+	activator = null
+	mychild.revive(full_heal = TRUE, admin_revive = TRUE)
+	if(boosted)
+		mychild.maxHealth = mychild.maxHealth * 0.5
+		mychild.health = mychild.maxHealth
+		if(times_won == 1)
+			SEND_SOUND(mychild, sound('sound/effects/magic.ogg'))
+			to_chat(mychild, "<span class='boldwarning'>As the life in the activator's eyes fade, the forcefield around you dies out and you feel your power subside.\nDespite this inferno being your home, you feel as if you aren't welcome here anymore.\nWithout any guidance, your purpose is now for you to decide.</span>")
+			to_chat(mychild, "<b>Your max health has been halved, but can now heal by standing on your tumor.  Note, it's your only way to heal.\nBear in mind, if anyone interacts with your tumor, you'll be resummoned here to carry out another fight.  In such a case, you will regain your full max health.\nAlso, be weary of your fellow inhabitants, they likely won't be happy to see you!</b>")
+			
+/obj/item/tumor_shard
+	name = "tumor shard"
+	desc = "A strange, sharp, crystal shard from an odd tumor on Lavaland.  Stabbing the corpse of a lavaland elite with this will revive them, assuming their soul still lingers.  Revived lavaland elites only have half their max health, but are completely loyal to their reviver."
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_state = "crevice_shard"
+	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
+	item_state = "screwdriver_head"
+	throwforce = 5
+	w_class = WEIGHT_CLASS_SMALL
+	throw_speed = 3
+	throw_range = 5
+	
+/obj/item/tumor_shard/afterattack(atom/target, mob/user, proximity_flag)
+	. = ..()
+	if(istype(target, /mob/living/simple_animal/hostile/asteroid/elite) && proximity_flag)
+		var/mob/living/simple_animal/hostile/asteroid/elite/E = target
+		if(E.stat == DEAD && E.sentience_type == SENTIENCE_BOSS)
```

Comment:
is there every a time where elite's won't have `SENTIENCE_BOSS`? IF not, then this latter check is kinda redundant since you're only allowing this to work on elites anyway.

Reply:
Once they are revived once with a shard, their sentience type switches so they can get laz'd properly.